backend/otel/otel.go

```diff
@@ -81,21 +81,24 @@ func setHighlightAttributes(attrs map[string]any, projectID, sessionID, requestI
 	}
 }
 
-func getLogRow(ctx context.Context, ts time.Time, lvl, projectID, sessionID, traceID, spanID string, excMessage string, resourceAttributes, spanAttributes, eventAttributes map[string]any, source modelInputs.LogSource) *clickhouse.LogRow {
+func getLogRow(ctx context.Context, ts time.Time, lvl, projectID, sessionID, traceID, spanID string, excMessage string, resourceAttributes, spanAttributes, eventAttributes map[string]any, source modelInputs.LogSource) (*clickhouse.LogRow, error) {
+	projectIDInt, err := clickhouse.ProjectToInt(projectID)
+
+	if err != nil {
+		return nil, err
+	}
+
 	return clickhouse.NewLogRow(
-		clickhouse.LogRowPrimaryAttrs{
-			TraceId:         traceID,
-			SpanId:          spanID,
-			SecureSessionId: sessionID,
-		},
-		clickhouse.WithTimestamp(ts),
+		ts, uint32(projectIDInt),
+		clickhouse.WithTraceID(traceID),
+		clickhouse.WithSpanID(spanID),
+		clickhouse.WithSecureSessionID(sessionID),
 		clickhouse.WithBody(ctx, excMessage),
 		clickhouse.WithLogAttributes(ctx, resourceAttributes, spanAttributes, eventAttributes, source == modelInputs.LogSourceFrontend),
-		clickhouse.WithProjectIDString(projectID),
```

Comment:
Previously, when `WithProjectIDString` had an error, we should just not write the project id for a log. It makes no sense to store a log with a project id. Hence, that could have masked a bug.

See the code on the right ➡️ 
The code that could error was `ProjectToInt`. If there's an error in that code, we bubble it up, log an error message, and skip writing this log.

Reply:
Looking into how `ProjectToInt` works, it appears it can error when someone passes in an invalid verbose project id:

https://github.com/highlight/highlight/blob/1cae712a59e123a3608d3701edaef72bb621aa3a/backend/clickhouse/log_row.go#L26-L30

Since the old code wouldn't actually set a project id for these logs, it defaults to 0 (clickhouse [defaults](https://clickhouse.com/docs/en/sql-reference/statements/create/table#default_values) to `0` for numeric columns).

I did a quick check on prod and it looks like there's a lot of logs with the bad data:

```sql
select count(*) from logs where ProjectId = 0;
=> 638693
```